### DISCUSSION TITLE
Localization not working

### DISCUSSION THREAD

**Discussion by bighamj** [iTwin/itwinjs-core#4892] [Q&A] [answered]
Working within an existing react application and having an issue with the labels on UI elements displaying Keys rather than the translations (just using English).

I've tried in our app as well as in a repo built by the iTwin viewer template. Both return something like this:

![image](https://user-images.githubusercontent.com/66083892/211386985-0859777f-d317-4b5e-8604-5cd4654ae8cd.png)

Looking at network tab shows the various items being loaded. In the build folder, there is indeed each of these files containing JSON. 

![image](https://user-images.githubusercontent.com/66083892/211387360-1fbe79fb-0237-4302-9ae7-312a8fe0ee14.png)

**Comment by aruniverse** [comment, marked as answer]
I see in the req url, it's looking for the locales file under `localhost:3000/bentley`. By default static assets (like locale files) are fetched relative to the current route. But if you're using our default template/config, the static assets will be served from `localhost:3000` **not** `localhost:3000/bentley`. 

To get around this you can set a variable on the window, `ITWIN_VIEWER_HOME` to the origin like so:
```ts
window.ITWIN_VIEWER_HOME = window.location.origin;
```
I'd recommend setting that in your index.tsx

**Reply by bighamj** [reply]
Perfect. Easy fix, thanks!
